apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -46,8 +57,17 @@ impl Drop for Oracle {
     fn drop(&mut self) {
         // send an abort signal to shut down the
         // rest of the ledger gracefully
-        let abort = self.abort.take().unwrap();
-        let _ = abort.send(());
+        match self.abort.take().unwrap().send(()) {
+            Ok(()) => tracing::info!("Oracle sent abort signal"),
+            Err(()) => {
+                // this isn't necessarily an issue as the ledger may have shut
+                // down first
+                tracing::debug!(
+                    "Oracle was unable to send an abort signal as the abort \
+                     channel was already closed"
+                )
+            }
+        };
```

Comment:
I think this is impossible in practice. we wait for every sub-task in the ledger to exit gracefully, with the `try_join!()` macro.

Reply:
Do you mean you think it's impossible for the oracle to send an abort signal? When running the ledger and pressing <kbd>Ctrl+C</kbd> I do only ever see the second message (unable to send abort signal). I think the flow of control for the oracle will have to change soon anyway in that it should be controlled from the ledger rather than as a "top level" task, as it will need access to parameters from blockchain storage like `MIN_CONFIRMATIONS` and will probably need to be epoched in some way (in that it should perhaps only run for certain epochs when a node is an active validator, and in each epoch it might have different parameters and also should check a different range of Ethereum blocks)